exercises/srikanth_ravipati/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+#include <fstream>
+#include "processLine.h"
+
+
+int main(int argc, char** argv)
+{
+
+    if (argc < 2){
+       throw std::runtime_error("Atleast one argument (filename) should be passed. USAGE: ./a.out inputFileName");
+       return 1;
+    } else {
+    std::cout << "File to be processed: "  << argv[1] << "\n";
+    }
+
+    auto nameInFile = argv[1];
+
+    std::vector<char> delimiters = {'\'','\"','.',',','?','!','(',')',':',' ','-','}'};
+    std::map<std::string,int> mapWordCounter;
+
+    std::ifstream inputFile;
+    inputFile.open(nameInFile, std::ios::in);
+
+    std::string line;
+    if (inputFile.is_open()){
+        while (getline(inputFile, line)){
+              processLine(line, delimiters, mapWordCounter);
+        }
+    }
+    inputFile.close();
+
+    std::vector<std::pair<std::string,int>> vecWordCounter;
+    for (auto& itr : mapWordCounter){
+        vecWordCounter.push_back(itr);
+    }
+
+    sort(vecWordCounter.begin(), vecWordCounter.end(), [](auto& a, auto& b){ return a.second > b.second;});
```

Comment:
We *should* define we are using the STL when intending to otherwise it has to guess what you meant and not only may pick a different function than intended but may take longer at compilation time (Negligible on it's own but if it happens 10,000 times in a code base it builds up etc etc) https://en.cppreference.com/w/cpp/language/adl
```suggestion
    std::sort(vecWordCounter.begin(), vecWordCounter.end(),
         [](auto &a, auto &b) { return a.second > b.second; });
```